lib/faraday/middleware.rb

```diff
@@ -7,9 +7,43 @@ class Middleware
 
     attr_reader :app, :options
 
+    DEFAULT_OPTIONS = {}.freeze
+
     def initialize(app = nil, options = {})
       @app = app
-      @options = options
+      @options = self.class.default_options.merge(options)
+    end
+
+    class << self
+      # Faraday::Middleware::default_options= allows user to set default options at the Faraday::Middleware
+      # class level.
+      #
+      # @example Set the Faraday::Response::RaiseError option, `include_request` to `false`
+      # my_app/config/initializers/my_faraday_middleware.rb
+      #
+      # Faraday::Response::RaiseError.default_options = { include_request: false }
+      #
+      def default_options=(options = {})
+        validate_default_options(options)
+        @default_options = default_options.merge(options)
```

Comment:
This method reads/writes from a class-level variable, so it's not thread-safe.
Could you please wrap its implementation into a `Monitor` object?

Reply:
Thanks for the feedback! ✅